db/background_mgr_dcp_client.go

```diff
@@ -0,0 +1,110 @@
+//  Copyright 2024-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package db
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbaselabs/rosmar"
+
+	sgbucket "github.com/couchbase/sg-bucket"
+)
+
+type backgroundMgrDcpClient interface {
+	// Start returns an error and a channel to indicate when the DCPClient is done. If Start returns an error, DCPClient.Close() needs to be called.
+	Start() (doneChan chan error, err error)
+	// Close is used externally to stop the DCP client. If the client was already closed due to error, returns that error
+	Close() error
+	// GetMetadata returns metadata for all vbuckets
+	GetMetadata() []base.DCPMetadata
+}
+
+// backgroundManagerDcpClientOptions are the dcp client options needed for creation of a background task dcp client
+type backgroundManagerDcpClientOptions struct {
```

Comment:
I think this is perhaps an easier cleanup and came because the initial implementation of `DCPClient` and the upgrade from gocb v1 to v2 happened before collection support was added.

For `DCPClient`, the collection ids are assembled https://github.com/couchbase/sync_gateway/blob/d9a47e914ffa1dc7f995213339de2c4bb085bd52/base/gocb_dcp_feed.go#L58 but I think this code could pass in `Scopes           map[string][]string // Collection names to stream - map keys are scopes` to `DCPClientOptions` and we wouldn't have to do specify this twice and can just move the determination of collection IDs (needed for gocbcore.OpenStreamFilterOptions.CollectionIDs).